### DISCUSSION TITLE
Node name is too long for the grid naming convention

### DISCUSSION THREAD

**Discussion by medhay36** [acroucher/PyTOUGH#44] [Q&A] [answered]
Hello dear scientists,

I am encountering an error while generating a mesh of a radioactive waste repository at full scale.
"Node name is too long for the grid naming convention.")
    904 return name

NamingConventionError: Node name is too long for the grid naming convention.

How to change the naming convention to allow bigger mesh size needed for some large scale simulations ? 

![image](https://user-images.githubusercontent.com/122643918/226487051-87e25e53-35f7-4fa0-b1ca-fa39ae7c3a71.png)

**Comment by acroucher** [comment, marked as answer]
This means there are not enough unique node names available for the nodes in your mesh. There are a few things you can do:

- make sure you are using the most appropriate naming convention (see appendix A.4 in the user guide). If your mesh has a lot of columns and not so many layers, you probably want to use convention 0 (3 characters for column, 2 digits for layer).
- use a bigger character set for your node/block names. By default it uses only lowercase letters, so you can make it use uppercase ones as well, and even add other things like punctuation marks as well if you want.

e.g. if you are creating a rectangular mesh:

```python
from mulgrids import *
# define xblocks, yblocks, zblocks lists here...
geo = mulgrid().rectangular(xblocks, yblocks, zblocks,
                            convention = 0,
                            chars = ascii_lowercase + ascii_uppercase)
```

**Reply by medhay36** [reply]
Thank you very much dear professor. Problem solved with changing naming convention.
For other users, just be patient  if you want to generate VTK file (process may take several minutes if the mesh is very big) 

![image](https://user-images.githubusercontent.com/122643918/226843887-28f92e23-b0b7-4e20-ba05-0dcf7c80109d.png)
